--- bin/../src3/py2neo/neo4j.py	(original)
+++ bin/../src3/py2neo/neo4j.py	(refactored)
@@ -19,8 +19,8 @@
 """
 
 
-import rest
-from urllib import quote
+from . import rest
+from urllib.parse import quote
 
 
 __version__   = "0.96"
@@ -626,7 +626,7 @@
 			td = td.uniqueness(uniqueness)
 		if relationships:
 			for relationship in (relationships or []):
-				if isinstance(relationship, (str,unicode)):
+				if isinstance(relationship, str):
 					td = td.relationships(relationship)
 				else:
 					td = td.relationships(*relationship)
